flight/tests/misc_math/unittest.cpp

```diff
@@ -350,3 +354,389 @@ TEST_F(LinearInterpolate, CollectiveCurve1to1) {
+     /* And crud */
+     0, 0, 0, 0, 0, 1, 1, 0,
+     0, 0, 0, 0, 0, -2, -2, 0,
+  };
+
+  float lotsquad_mixer[10 * 8] = {
+     0.5f,  0.5f,  0.5f,  1.0f, 0, 0, 0, 0,
+    -0.5f, -0.5f,  0.5f,  1.0f, 0, 0, 0, 0,
+     0.5f, -0.5f, -0.5f,  1.0f, 0, 0 ,0, 0,
+    -0.5f,  0.5f, -0.5f,  1.0f, 0, 0, 0, 0,
+     0.5f,  0.5f,  0.5f,  1.0f, 0, 0, 0, 0,
+    -0.5f, -0.5f,  0.5f,  1.0f, 0, 0, 0, 0,
+     0.5f, -0.5f, -0.5f,  1.0f, 0, 0 ,0, 0,
+    -0.5f,  0.5f, -0.5f,  1.0f, 0, 0, 0, 0,
+     0.5f,  0.5f,  0.5f,  1.0f, 0, 0, 0, 0,
+    -0.5f, -0.5f,  0.5f,  1.0f, 0, 0, 0, 0,
+  };
+
+  float octo_mixer[10 * 8] = {
+     0.707f,  0.707f,  0.5f,  1.0f, 0, 0, 0, 0,
+    -0.707f, -0.707f,  0.5f,  1.0f, 0, 0, 0, 0,
+     0.707f, -0.707f, -0.5f,  1.0f, 0, 0 ,0, 0,
+    -0.707f,  0.707f, -0.5f,  1.0f, 0, 0, 0, 0,
+     1.0f,  0.0f,  0.5f,  1.0f, 0, 0, 0, 0,
+    -1.0f,  0.0f, -0.5f,  1.0f, 0, 0 ,0, 0,
+     0.0f, -1.0f,  0.5f,  1.0f, 0, 0, 0, 0,
+     0.0f,  1.0f, -0.5f,  1.0f, 0, 0, 0, 0,
+     0, 0, 0, 1, 0, 0, 0, 0,
+     0, 0, 0, 0, 0, -2, -2, 0,
+  };
+
+  float elevon_mixer[10 * 8] = {
+     0.5f,    0.5f,    0.0f,  0.0f, 0, 0, 0, 0,
+    -0.492f,  0.5f,    0.0f,  0.0f, 0, 0, 0, 0,
+     0.0f,    0.0f,    0.0f,  1.0f, 0, 0, 0, 0,
+     0.0f,    0.0f,    1.0f,  0.0f, 0, 0, 0, 0,
+     0.0f,    0.0f,    0.0f,  0.0f, 1, 1, 0, 0,
+     0.0f,    0.0f,    0.0f,  0.0f, -1, 1, 0, 0,
+     0.0f,    0.0f,    0.0f,  0.0f, 1, 1, 0, 0,
+     0.0f,    0.0f,    0.0f,  0.0f, -1, 1, 0, 0,
+  };
+
+  float inv_motor_mixer[8 * 10];
+  float temporary[10 * 10];
+  float should_be_motor_mixer[10 * 8];
+
+  EXPECT_TRUE(matrix_pseudoinv(quad_mixer, inv_motor_mixer, 10, 8));
+
+  matrix_mul_check(quad_mixer, inv_motor_mixer, temporary, 10, 8, 10);
+
+  matrix_mul_check(temporary, quad_mixer, should_be_motor_mixer, 10, 10, 8);
+
+  for (int i=0; i<10*8; i++) {
+    EXPECT_NEAR(quad_mixer[i], should_be_motor_mixer[i], eps);
+  }
+
+  EXPECT_TRUE(matrix_pseudoinv(hexacoax_mixer, inv_motor_mixer, 10, 8));
+
+  matrix_mul_check(hexacoax_mixer, inv_motor_mixer, temporary, 10, 8, 10);
+
+  matrix_mul_check(temporary, hexacoax_mixer, should_be_motor_mixer, 10, 10, 8);
+
+  for (int i=0; i<10*8; i++) {
+    EXPECT_NEAR(hexacoax_mixer[i], should_be_motor_mixer[i], eps);
+  }
+
+  for (int j=0; j<500; j++) {
+    /* Round trip a few times */
+
+    EXPECT_TRUE(matrix_pseudoinv(should_be_motor_mixer, inv_motor_mixer, 10, 8));
+    matrix_mul_check(should_be_motor_mixer, inv_motor_mixer, temporary, 10, 8, 10);
+
+    matrix_mul_check(temporary, hexacoax_mixer, should_be_motor_mixer, 10, 10, 8);
+
+    for (int i=0; i<10*8; i++) {
+      EXPECT_NEAR(hexacoax_mixer[i], should_be_motor_mixer[i], eps);
+    }
+  }
+
+  EXPECT_TRUE(matrix_pseudoinv(lotsquad_mixer, inv_motor_mixer, 10, 8));
+
+  matrix_mul_check(lotsquad_mixer, inv_motor_mixer, temporary, 10, 8, 10);
+
+  matrix_mul_check(temporary, lotsquad_mixer, should_be_motor_mixer, 10, 10, 8);
+
+  for (int i=0; i<10*8; i++) {
```

Comment:
I'm adding spaces to the extent where it adds readability   ... e.g  I get things like `i < 10*8` because otherwise things get too spacey.